### PR TITLE
Next stage of (error) refactoring

### PR DESCRIPTION
**Pull request by grzkv**
- not-found during `/find` request is not considered an error anymore both on `api` and `zipper`
- added metric for not founds on `api`
- fixed double-counting of requests on `zipper` `/find` in case of not-found
- removed obsolete fan-ins in `api` handlers
- broke `/render` handler in *api* into smaller pieces

### CODE REVIEW

File: app/carbonapi/http_handlers.go [hunk 1]
```diff
@@ -428,37 +359,143 @@ func (app *App) renderHandler(w http.ResponseWriter, r *http.Request) {
...
+	switch form.format {
 	case jsonFormat:
 		if maxDataPoints, _ := strconv.Atoi(r.FormValue("maxDataPoints")); maxDataPoints != 0 {
 			types.ConsolidateJSON(maxDataPoints, results)
 		}
 
 		body = types.MarshalJSON(results)
 	case protobufFormat, protobuf3Format:
-		body, err = types.MarshalProtobuf(results)
+		body, err := types.MarshalProtobuf(results)
```
Comment (gksinghjsr) [inline app/carbonapi/http_handlers.go]: Won't this shadow the variable created in outer scope and create issues?

Comment (grzkv) [reply app/carbonapi/http_handlers.go]: This will shadow, but I don't see problem with that.

Comment (gksinghjsr) [reply app/carbonapi/http_handlers.go]: https://play.golang.org/p/qkUigxU9eFN

Comment (grzkv) [reply app/carbonapi/http_handlers.go]: Yes, but what's the problem? Shadowing value is only used inside single case as expected.

Comment (gksinghjsr) [reply app/carbonapi/http_handlers.go]: body would return nil in case format is one of protobufFormat, protobuf3Format

Comment (grzkv) [reply app/carbonapi/http_handlers.go]: If you talk about body, then yes, it is a problem.

**Pull request merged**